### PR TITLE
Xline cluster test

### PR DESCRIPTION
**Pull request by themanforfree**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
Add tests for xline cluster server
* what changes does this pull request make?
Add tests for xline cluster server
* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
no

### CODE REVIEW

File: xline/tests/cluster_test.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+async fn xline_update_node() -> Result<(), Box<dyn Error>> {
+    let mut cluster = Cluster::new(3).await;
+    cluster.start().await;
+    let mut cluster_client = Client::connect(cluster.addrs(), ClientOptions::default())
+        .await?
+        .cluster_client();
+    let old_list_res = cluster_client
+        .member_list(MemberListRequest::new(false))
+        .await?;
+    println!("{:?}", old_list_res.members);
```
Comment (Phoenix500526) [inline xline/tests/cluster_test.rs:66]: Please remove the debug print.

File: xline/tests/cluster_test.rs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    let new_listener = TcpListener::bind("0.0.0.0:0").await?;
+    let new_addrs = vec![new_listener.local_addr()?.to_string()];
+    let update_req = MemberUpdateRequest::new(update_id, new_addrs.clone());
+    let update_res = cluster_client.member_update(update_req).await?;
+    assert_eq!(update_res.members.len(), 3);
+    sleep(Duration::from_secs(3)).await;
+    let new_list_res = cluster_client
+        .member_list(MemberListRequest::new(false))
+        .await?;
+    println!("{:?}", new_list_res.members);
```
Comment (Phoenix500526) [inline xline/tests/cluster_test.rs:78]: Same as above

Comment (bsbds) [inline xline/tests/cluster_test.rs:78]: Please remove these prints

File: curp/src/server/cmd_worker/mod.rs [hunk 3]
```diff
@@ -163,6 +163,8 @@ async fn worker_as<
             true
         }
         EntryData::ConfChange(ref conf_change) => {
+            ce.trigger_id(conf_change.id());
+            ce.trigger_index(entry.index);
```
[the remark this answers is not in the document]
Comment (themanforfree) [reply curp/src/server/cmd_worker/mod.rs]: Fixed

File: curp/src/server/raw_curp/state.rs [hunk 4]
```diff
@@ -287,8 +287,8 @@ impl Config {
     pub(super) fn remove(&mut self, id: ServerId) -> bool {
         let res1 = self.majority_config.voters.remove(&id);
         let res2 = self.learners.remove(&id);
-        debug_assert_ne!(
-            res1, res2,
+        debug_assert!(
+            !(res1 && res2),
```
[the remark this answers is not in the document]
Comment (themanforfree) [reply curp/src/server/raw_curp/state.rs]: Fixed

File: xline/tests/cluster_test.rs (hunk 1 above)
Comment (themanforfree) [reply xline/tests/cluster_test.rs]: Fixed

File: xline-test-utils/src/lib.rs [hunk 5]
```diff
@@ -42,20 +38,18 @@ impl Cluster {
         }
         let all_members = listeners
             .iter()
-            .map(|(i, l)| (format!("server{}", i), l.local_addr().unwrap().to_string()))
+            .map(|(i, l)| (*i, l.local_addr().unwrap().to_string()))
```
[the remark this answers is not in the document]
Comment (themanforfree) [reply xline-test-utils/src/lib.rs:41]: Same as above

File: xline/tests/cluster_test.rs (hunk 2 above)
Comment (themanforfree) [reply xline/tests/cluster_test.rs]: Removed

File: xline-test-utils/src/lib.rs [hunk 6]
```diff
@@ -22,15 +20,13 @@ pub struct Cluster {
...
+    all_members: HashMap<usize, String>,
     /// Client of cluster
     client: Option<Client>,
-    /// Xline servers
-    servers: Vec<XlineServer>,
     /// Cluster size
     size: usize,
     /// storage paths
-    paths: Vec<PathBuf>,
+    paths: HashMap<usize, PathBuf>,
```
[the remark this answers is not in the document]
Comment (themanforfree) [reply xline-test-utils/src/lib.rs:29]: Same as above, Using vec will disturb the relationship between index and node when removing nodes.

File: xline-test-utils/src/lib.rs [hunk 7]
```diff
@@ -22,15 +20,13 @@ pub struct Cluster {
     /// listeners of members
     listeners: BTreeMap<usize, TcpListener>,
     /// address of members
-    all_members: HashMap<String, String>,
+    all_members: HashMap<usize, String>,
```
[the remark this answers is not in the document]
Comment (themanforfree) [reply xline-test-utils/src/lib.rs:23]: In the old implementation, String is all `format! {"server{idx}"}`, so I want to use usize to specify a node

**Review by Phoenix500526 — APPROVED** [review]
LGTM

File: xline/src/server/command.rs [hunk 8]
```diff
@@ -305,14 +305,10 @@ where
     fn prepare(
         &self,
         cmd: &Command,
-        index: LogIndex,
+        _index: LogIndex,
```
Comment (bsbds) [inline xline/src/server/command.rs:308]: Please remove the index if it not needed

File: xline/src/server/command.rs [hunk 9]
```diff
@@ -335,21 +331,13 @@ where
     async fn execute(
         &self,
         cmd: &Command,
-        index: LogIndex,
+        _index: LogIndex,
```
Comment (bsbds) [inline xline/src/server/command.rs:334]: same as above

File: xline/src/server/command.rs (hunk 8 above)
Comment (themanforfree) [reply xline/src/server/command.rs]: Removed

File: xline/src/server/command.rs (hunk 9 above)
Comment (themanforfree) [reply xline/src/server/command.rs]: Removed

**Review by Phoenix500526 — APPROVED** [review]
LGTM

**Review by bsbds — APPROVED** [review]

**Pull request merged**